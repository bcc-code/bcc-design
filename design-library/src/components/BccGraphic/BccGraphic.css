@layer components {
  .bcc-graphic {
    @apply relative w-full cursor-pointer;
    @apply h-40 md:h-52 lg:h-72;
  }

  .bcc-graphic-open-in {
    @apply absolute bottom-4 right-4 z-10 px-3 py-2;
    @apply rounded-md border border-gray-400 bg-white transition-all dark:border-gray-600 dark:bg-black;
  }

  .bcc-graphic-open-in-icon {
    @apply h-4 w-4;
  }
  .bcc-graphic:hover .bcc-graphic-open-in {
    @apply bg-gray-200;
  }
  .bcc-graphic:hover .bcc-graphic-check-icon {
    @apply text-gray-200;
  }
  .bcc-graphic-check-icon {
    @apply absolute right-4 top-4 z-10 h-6 w-6 text-white transition-all;
  }

  .bcc-graphic-poster {
    @apply flex h-full w-full items-center justify-center;
    @apply  bg-gradient-to-tr from-brand-500 to-brand-600 bg-cover bg-center;
    /* Default rounding */
    @apply rounded-xl;
  }

  /* Rounding */
  .bcc-graphic-poster-rounding-md {
    @apply rounded-md;
  }
}
