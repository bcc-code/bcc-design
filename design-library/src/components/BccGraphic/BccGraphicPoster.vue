<script lang="ts">
export const ratioClasses = {
  ultraWide: "42.85%", // 21:9
  wide: "56.25%", // 16:9
  landscape: "75%", // 4:3
  square: "100%", // 1:1
  portrait: "133.33%", // 3:4
};

export const roundingClasses = {
  none: "",
  sm: "rounded-sm",
  base: "rounded",
  md: "rounded-md",
  xl: "rounded-xl",
};
</script>

<script setup lang="ts">
import { ref } from "vue";

type Props = {
  bannerSrc?: string;
  logoSrc?: string;
  rounding?: keyof typeof roundingClasses;
  ratio?: keyof typeof ratioClasses;
};

withDefaults(defineProps<Props>(), {
  rounding: "xl",
  ratio: "wide",
});

const loadedBanner = ref(false);
const loadedLogo = ref(false);
</script>

<template>
  <div
    class="bcc-graphic-poster"
    :style="`padding-bottom: ${ratioClasses[ratio]}`"
    :class="roundingClasses[rounding]"
  >
    <template v-if="bannerSrc">
      <img
        v-show="loadedBanner"
        @load="loadedBanner = true"
        :src="bannerSrc"
        defer
        class="bcc-graphic-poster-banner"
        :class="roundingClasses[rounding]"
      />
      <div
        v-show="!loadedBanner"
        class="bcc-graphic-poster-banner bcc-graphic-poster-banner--loading"
        :class="roundingClasses[rounding]"
      ></div>
    </template>

    <template v-if="logoSrc">
      <img
        v-show="loadedLogo"
        @load="loadedLogo = true"
        :src="logoSrc"
        defer
        class="bcc-graphic-poster-logo"
      />
      <div
        v-show="!loadedLogo"
        class="bcc-graphic-poster-logo bcc-graphic-poster-logo--loading"
      ></div>
    </template>
  </div>
</template>
