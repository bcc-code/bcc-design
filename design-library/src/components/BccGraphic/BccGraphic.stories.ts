import BccGraphic from "./BccGraphic.vue";

import type { Meta, StoryFn } from "@storybook/vue3";

export default {
  title: "Components/BccGraphic",
  component: BccGraphic,
  argTypes: {
    rounding: {
      options: ["md", "xl"],
      control: { type: "radio" },
    },
  },
} as Meta<typeof BccGraphic>;

const Template: StoryFn<typeof BccGraphic> = (args) => ({
  components: { BccGraphic },
  setup() {
    return { args };
  },
  template: `
    <div class="flex flex-col gap-y-2">
      <BccGraphic v-bind="args" />
    </div>
  `,
});

export const Example = Template.bind({});
Example.args = {
  bannerSrc: "https://event.bcc.no/wp-content/uploads/2023/10/Background-7.png",
  logoSrc: "https://event.bcc.no/wp-content/uploads/2023/11/BUK4_Logo-Main.svg",
  rounding: "xl",
  checked: false,
  linkOut: false,
};
Example.parameters = {
  docs: {
    source: {
      language: "html",
      code: `
<BccGraphic bannerSrc="https://event.bcc.no/..." logoSrc="https://event.bcc.no/..." />
    `,
    },
  },
};

/**
 * A `bannerSrc` or `logoSrc` is not needed, by default this will just display a brand background. Combine with other props to create a more interesting look.
 */
export const Default: StoryFn<typeof BccGraphic> = () => ({
  components: { BccGraphic },
  template: `
    <BccGraphic />
  `,
});

/**
 * Display an icon to indicate there is a link on the graphic by setting the `linkOut` prop. Add an `@click` handler to handle the link.
 */
export const WithLinkOut: StoryFn<typeof BccGraphic> = () => ({
  components: { BccGraphic },
  template: `
    <BccGraphic linkOut />
  `,
});

/**
 * Set the `checked` prop to indicate a checked state on the banner.
 */
export const Checked: StoryFn<typeof BccGraphic> = () => ({
  components: { BccGraphic },
  template: `
    <BccGraphic checked />
  `,
});
