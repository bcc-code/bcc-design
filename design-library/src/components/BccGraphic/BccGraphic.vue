<script setup lang="ts">
import { OpenInNewIcon, CheckCircleFillIcon } from "@bcc-code/icons-vue";
import BccGraphicPoster from "./BccGraphicPoster.vue";

type Props = {
  bannerSrc?: string;
  logoSrc?: string;
  checked?: boolean;
  linkOut?: boolean;
  rounding?: "md" | "xl";
};

withDefaults(defineProps<Props>(), {
  rounding: "xl",
  checked: false,
  linkOut: false,
});
</script>

<template>
  <div class="bcc-graphic">
    <div class="bcc-graphic-open-in" v-if="linkOut">
      <OpenInNewIcon class="bcc-graphic-open-in-icon" />
    </div>
    <CheckCircleFillIcon class="bcc-graphic-check-icon" v-if="checked" />
    <BccGraphicPoster :bannerSrc="bannerSrc" :logoSrc="logoSrc" :rounding="rounding" />
  </div>
</template>
