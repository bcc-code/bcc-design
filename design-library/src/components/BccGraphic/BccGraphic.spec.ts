import { describe, it, expect } from "vitest";

import { mount } from "@vue/test-utils";
import BccGraphic from "./BccGraphic.vue";

const bannerSrc = "https://event.bcc.no/wp-content/uploads/2023/10/Background-7.png";
const logoSrc = "https://event.bcc.no/wp-content/uploads/2023/11/BUK4_Logo-Main.svg";

describe("BccGraphic", () => {
  it("renders graphic", async () => {
    const wrapper = mount(BccGraphic, {
      props: {
        bannerSrc,
        logoSrc,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders checked", async () => {
    const wrapper = mount(BccGraphic, {
      props: {
        bannerSrc,
        logoSrc,
        checked: true,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders link out", async () => {
    const wrapper = mount(BccGraphic, {
      props: {
        bannerSrc,
        logoSrc,
        linkOut: true,
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("renders rounding", async () => {
    const wrapper = mount(BccGraphic, {
      props: {
        logoSrc,
        rounding: "md",
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
