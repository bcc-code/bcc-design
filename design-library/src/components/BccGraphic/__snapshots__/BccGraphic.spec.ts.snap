// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BccGraphic > renders checked 1`] = `
"<div class=\\"bcc-graphic-wrapper\\">
  <!--v-if--><svg aria-hidden=\\"true\\" viewBox=\\"0 -960 960 960\\" fill=\\"currentColor\\" class=\\"bcc-graphic-check-icon\\">
    <path d=\\"m424-408-86-86q-11-11-28-11t-28 11q-11 11-11 28t11 28l114 114q12 12 28 12t28-12l226-226q11-11 11-28t-11-28q-11-11-28-11t-28 11L424-408Zm56 328q-83 0-156-31.5T197-197q-54-54-85.5-127T80-480q0-83 31.5-156T197-763q54-54 127-85.5T480-880q83 0 156 31.5T763-763q54 54 85.5 127T880-480q0 83-31.5 156T763-197q-54 54-127 85.5T480-80Z\\"></path>
  </svg>
  <div class=\\"relative h-full\\">
    <div class=\\"bcc-graphic-poster\\" style=\\"background-image: url(https://event.bcc.no/wp-content/uploads/2023/10/Background-7.png); border-radius: 12px;\\"><img class=\\"h-3/4\\" src=\\"https://event.bcc.no/wp-content/uploads/2023/11/BUK4_Logo-Main.svg\\"></div>
  </div>
</div>"
`;

exports[`BccGraphic > renders graphic 1`] = `
"<div class=\\"bcc-graphic-wrapper\\">
  <!--v-if-->
  <!--v-if-->
  <div class=\\"relative h-full\\">
    <div class=\\"bcc-graphic-poster\\" style=\\"background-image: url(https://event.bcc.no/wp-content/uploads/2023/10/Background-7.png); border-radius: 12px;\\"><img class=\\"h-3/4\\" src=\\"https://event.bcc.no/wp-content/uploads/2023/11/BUK4_Logo-Main.svg\\"></div>
  </div>
</div>"
`;

exports[`BccGraphic > renders link out 1`] = `
"<div class=\\"bcc-graphic-wrapper\\">
  <div class=\\"bcc-graphic-open-in\\"><svg aria-hidden=\\"true\\" viewBox=\\"0 -960 960 960\\" fill=\\"currentColor\\" class=\\"bcc-graphic-open-in-icon\\">
      <path d=\\"M200-120q-33 0-56.5-23.5T120-200v-560q0-33 23.5-56.5T200-840h240q17 0 28.5 11.5T480-800q0 17-11.5 28.5T440-760H200v560h560v-240q0-17 11.5-28.5T800-480q17 0 28.5 11.5T840-440v240q0 33-23.5 56.5T760-120H200Zm560-584L416-360q-11 11-28 11t-28-11q-11-11-11-28t11-28l344-344H600q-17 0-28.5-11.5T560-800q0-17 11.5-28.5T600-840h240v240q0 17-11.5 28.5T800-560q-17 0-28.5-11.5T760-600v-104Z\\"></path>
    </svg></div>
  <!--v-if-->
  <div class=\\"relative h-full\\">
    <div class=\\"bcc-graphic-poster\\" style=\\"background-image: url(https://event.bcc.no/wp-content/uploads/2023/10/Background-7.png); border-radius: 12px;\\"><img class=\\"h-3/4\\" src=\\"https://event.bcc.no/wp-content/uploads/2023/11/BUK4_Logo-Main.svg\\"></div>
  </div>
</div>"
`;

exports[`BccGraphic > renders rounding 1`] = `
"<div class=\\"bcc-graphic-wrapper\\">
  <!--v-if-->
  <!--v-if-->
  <div class=\\"relative h-full\\">
    <div class=\\"bcc-graphic-poster\\" style=\\"border-radius: 6px;\\"><img class=\\"h-3/4\\" src=\\"https://event.bcc.no/wp-content/uploads/2023/11/BUK4_Logo-Main.svg\\"></div>
  </div>
</div>"
`;
